executor/point_get.go

```diff
@@ -336,7 +342,27 @@ func (e *PointGetExecutor) get(ctx context.Context, key kv.Key) ([]byte, error)
 		}
 		// fallthrough to snapshot get.
 	}
-	return e.snapshot.Get(ctx, key)
+
+	isLocked := e.tblInfo.IsLocked()
+	if !isLocked || e.tblInfo.Lock.Tp != model.TableLockRead { // if the lock is read or tale was unlock then snapshot get
```

Comment:
`e.tblInfo.Lock.Tp != model.TableLockRead` means what?

Reply:
There are three type lock, read lock, write lock and write local lock. 
We only query from cache with read lock.